### PR TITLE
MDX fix for Storybook + Developer Guides for internal use

### PR DESCRIPTION
**Pull request by spark33**
## ✍️ Proposed changes

image

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Style change (purely visual updates; if this is the case, attach a screenshot below!)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

## Feedback Levels

Please try to mark your comments with one of these levels

1. Praise (👏 🚀 🔥 🍾)
2. Opinions, Ideas, Suggestions (💡 👀)
3. Let’s discuss @ Code review (🤷 🤔 🧐 )
4. Doesn’t meet agreed upon standards (😬 ⚠️ ‼️)
5. Missing an edge case (🛑 🙅 ❌)
6. Missing core functionality (⛔️ ☣️ 🚧 🌋 💀)

(Note: Use ⌘-⌃-␣ to open the Emoji keyboard!)

### CODE REVIEW

File: stories/Folder-Structure.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { Meta } from '@storybook/addon-docs';
+import { H1 } from '@leafygreen-ui/typography';
+import BannerImage from '../.storybook/static/leafygreen-ui-banner.png';
+
+<Meta title="Developer Guide/Package Folder Structure" />
+
+### Folder Structure
```
Comment (brookescarlett) [inline stories/Folder-Structure.stories.mdx:7]: indicate that even packages that only export one directory have a subcomponent directory i.e. button/Button/Button.tsx

**Review by brookescarlett — COMMENTED** [review]
I appreciate this very much, and LGTM. @shaneeza and @TheSonOfThomp would love your stamp or feedback to, as this impacts everyone's work moving forward, so want to make sure that this is codified correctly.

File: stories/Folder-Structure.stories.mdx (hunk 1 above)
Comment (spark33) [reply stories/Folder-Structure.stories.mdx:7]: Added!

File: stories/Folder-Structure.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+- `<ComponentName>.tsx`
+- `<ComponentName>.spec.tsx**`
+- `<ComponentName>.stories.tsx**`
+- `any subcomponent directories`
+
+***only if the stories or tests pertain specifically to the component in this directory; meaning it may not be necessary for many internal subcomponents*
+
+The folder structure should be identical for all components - whether they are considered components that are exported to npm, or subcomponents contained within those packages. In other words, **all subcomponents are simply components**, and they should be structured identically to reflect that.
+
+Even if a package only has one component, it should have a subdirectory. For example, the path to the component file for `Badge` would be `/badge/src/Badge/Badge.tsx`. While this could be seen as clutter, it allows us to differentiate code that is relevant to the entire *package* from code that only pertains to the *component*. It is also more scalable as adding subcomponents in the future would simply involve adding a new directory, instead of having to move all files in the package.
```
Comment (TheSonOfThomp) [inline stories/Folder-Structure.stories.mdx:21]: I would disagree with this point. I would prefer all base component files to be in the root of `src` than to create an additional level of directory nesting. 

Most components won't have subcomponents, which makes this needlessly deep, and even for components with subcomponents, I don't see any issue with having the main component's files being in the src root.

i.e.
```
button/src
 - index.tsx
 - Button.tsx
 - Button.styles.tsx
 - Button.types.tsx
 - Button.stories.tsx
 - Button.spec.tsx
 - ButtonIcon
 - - index.tsx
 - - ButtonIcon.tsx
 - - etc...
```

File: stories/Folder-Structure.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import { H1 } from '@leafygreen-ui/typography';
+import BannerImage from '../.storybook/static/leafygreen-ui-banner.png';
+
+<Meta title="Developer Guide/Package Folder Structure" />
+
+### Folder Structure
+
+- `index.tsx`
+- `styles.ts`
+- `types.ts`
```
Comment (TheSonOfThomp) [inline stories/Folder-Structure.stories.mdx:11]: For the same reason we keep the component in `<ComponentName>.tsx` not index, I would advocate for using `<ComponentName>.styles.tsx`

File: stories/Folder-Structure.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+- `<ComponentName>.tsx`
+- `<ComponentName>.spec.tsx**`
+- `<ComponentName>.stories.tsx**`
+- `any subcomponent directories`
+
+***only if the stories or tests pertain specifically to the component in this directory; meaning it may not be necessary for many internal subcomponents*
+
```
Comment (TheSonOfThomp) [inline stories/Folder-Structure.stories.mdx:18]: This part is a bit unclear.  I would move this to the `Subcomponentization` section with a line saying something to the effect "if a file isn't needed for a subcomponent, don't add it"

File: stories/Folder-Structure.stories.mdx (hunk 3 above)
Comment (TheSonOfThomp) [reply stories/Folder-Structure.stories.mdx:11]: Additionally, this keeps names consistent with `*.stories.tsx` and `*.spec.tsx` files

File: stories/Folder-Structure.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+The folder structure should be identical for all components - whether they are considered components that are exported to npm, or subcomponents contained within those packages. In other words, **all subcomponents are simply components**, and they should be structured identically to reflect that.
+
+Even if a package only has one component, it should have a subdirectory. For example, the path to the component file for `Badge` would be `/badge/src/Badge/Badge.tsx`. While this could be seen as clutter, it allows us to differentiate code that is relevant to the entire *package* from code that only pertains to the *component*. It is also more scalable as adding subcomponents in the future would simply involve adding a new directory, instead of having to move all files in the package.
+
+### Subcomponentization
```
Comment (TheSonOfThomp) [inline stories/Folder-Structure.stories.mdx:23]: I think you just made up a word 😆

File: .storybook/preview-head.html [hunk 6]
```diff
@@ -41,8 +41,18 @@
...
 
+  #docs-root h1:not(:nth-child(1)),
+  #docs-root h2:not(:nth-child(1)),
+  #docs-root h3:not(:nth-child(1)),
+  #docs-root h4:not(:nth-child(1)),
+  #docs-root h5:not(:nth-child(1)),
+  #docs-root h6:not(:nth-child(1)) {
+    /* this acts like a margin-bottom for the following p tag in lieu of a previous sibling selector */
+    margin-top: 40px !important;
+  }
```
Comment (TheSonOfThomp) [inline .storybook/preview-head.html:52]: Double check this doesn't interfere with the Typography package

File: stories/Folder-Structure.stories.mdx (hunk 2 above)
Comment (shaneeza) [reply stories/Folder-Structure.stories.mdx:21]: I do think that this makes a component needlessly deep but I like the consistency of keeping everything in a subcomponent. I think the code above would look a bit neater if everything for `Button` was in its own folder.

File: stories/Folder-Structure.stories.mdx (hunk 5 above)
Comment (shaneeza) [reply stories/Folder-Structure.stories.mdx:23]: haha

File: stories/Folder-Structure.stories.mdx (hunk 2 above)
Comment (TheSonOfThomp) [reply stories/Folder-Structure.stories.mdx:21]: It may be a bit cleaner, but IMO it makes it harder to parse. If we have several folders within a `src` folder it makes it hard to parse which is the _main_ component, and which are subcomponents. Take the `SearchInput` component for instance. It takes a bit more mental effort to parse which is the main component, vs a case where the main component files are in the root of `src` 

```
search-input/src/
 - SearchInput
 - SearchResult
 - SearchResultMenu
 - SearchResultGroup
```

**Review by brookescarlett — APPROVED** [review]

**Comment by brookescarlett** [comment]
Will you copy this to the team resources wiki?
